### PR TITLE
The java.compiler system property is obsolete in jdk21+

### PR DESCRIPTION
**Pull request by pshipton**
Closes https://github.com/eclipse-openj9/openj9/issues/18430

Test build and VmArgumentTests passing results for jdk11, 21
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/459/
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/3118
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/3119/

### CODE REVIEW

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public class VmArgumentTests {
 	private static final String  MAPOPT_XJIT_COUNT_EQUALS = "-Xjit:count="+MIXED_MODE_THRESHOLD_VALUE;
 
 	private static final String JAVA_COMPILER = "JAVA_COMPILER";
-	private static final String JAVA_COMPILER_VALUE=System.getProperty("java.compiler");
+	private static final String JAVA_COMPILER_VALUE=VersionCheck.major() < 21 ? System.getProperty("java.compiler") : System.getProperty("openj9.compiler");
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:118]: Spaces around operators; suggest pushing the condition down:
```java
	private static final String JAVA_COMPILER_VALUE = System.getProperty(VersionCheck.major() < 21 ? "java.compiler" : "openj9.compiler");
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/nls/j9cl/j9jcl.nls [hunk 2]
```diff
@@ -588,3 +588,20 @@ J9NLS_JCL_CRIU_FAILED_TO_ENABLE_ALL_RESTORE_OPTIONS.explanation=NOTAG
 J9NLS_JCL_CRIU_FAILED_TO_ENABLE_ALL_RESTORE_OPTIONS.system_action=
 J9NLS_JCL_CRIU_FAILED_TO_ENABLE_ALL_RESTORE_OPTIONS.user_response=
 # END NON-TRANSLATABLE
+
+# Note: "java.compiler" is a system property name and should not be translated.
+J9NLS_JCL_JAVA_COMPILER_WARNING_XJIT=Setting the java.compiler system property is obsolete in version 21 and later, use -Xjit instead.
```
Comment (keithc-ca) [inline runtime/nls/j9cl/j9jcl.nls:593]: This should note that `-Xjit` is a command line parameter not to be translated.

Comment (pshipton) [reply runtime/nls/j9cl/j9jcl.nls:593]: The comment is unnecessary, the AI is smart enough not to translate it, but I will add it.

File: runtime/jcl/common/system.c [hunk 3]
```diff
@@ -551,6 +552,13 @@ systemPropertyIterator(char* key, char* value, void* userData)
 		return;
 	}
 
+#if JAVA_SPEC_VERSION >= 21
+	if (0 == strcmp("java.compiler", key)) {
+		PORT_ACCESS_FROM_ENV(env);
+		j9nls_printf(PORTLIB, J9NLS_WARNING, J9NLS_JCL_JAVA_COMPILER_WARNING);
```
[the remark this answers is not in the document]
Comment (pshipton) [reply runtime/jcl/common/system.c]: Before this PR, OpenJ9 looks for "jitc" and "j9jit29" and maps those to -Xjit. Anything else maps to -Xint.

[the remark this answers is not in the document]
Comment (keithc-ca) [reply runtime/jcl/common/system.c]: Ok, got it.

File: runtime/nls/j9cl/j9jcl.nls (hunk 2 above)
Comment (pshipton) [reply runtime/nls/j9cl/j9jcl.nls:594]: Doing translations is now done by machine and not by any person.

**Comment by keithc-ca** [comment]
Jenkins compile alinux jdk17,jdk21

**Comment by keithc-ca** [comment]
> new grinder on the jdk21

That was more than I was going to do. I think we're good.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**